src/util/implicitRoles/select.js

```diff
@@ -1,6 +1,21 @@
+import { getProp, getLiteralPropValue } from 'jsx-ast-utils';
+
 /**
- * Returns the implicit role for a select tag.
+ * Returns the implicit role for a select tag depending on attributes.
+ *
+ * @see https://www.w3.org/TR/html-aria/#el-select
  */
-export default function getImplicitRoleForSelect() {
-  return 'listbox';
+export default function getImplicitRoleForSelect(attributes) {
+  const multiple = getProp(attributes, 'multiple');
+  if (multiple && getLiteralPropValue(multiple)) {
+    return 'listbox';
+  }
+
+  const size = getProp(attributes, 'size');
+  const sizeValue = size && getLiteralPropValue(size);
+  if (sizeValue && (Number(sizeValue) > 1)) {
```

Comment:
```suggestion
  if (sizeValue > 1) {
```
fwiw i think this is equivalent and simpler